### ISSUE TITLE
Make `dist plan` handle mismatched project versions

### ISSUE THREAD

**Issue by duckinator** [axodotdev/cargo-dist#1554]
If there's packages with two different versions (let's say foo + bar 1.0.0 and baz 2.0.0), currently `dist plan` fails like this:

```
  × There are too many unrelated apps in your workspace to coherently Announce!
  help: Please either specify --tag, or give them all the same version
        
        Here are some options:
        
        --tag=v1.0.0 will Announce: foo, bar
        --tag=v2.0.0 will Announce: baz
        
        you can also request any single package with --tag=foo-v1.0.0
```

It would be useful to have it figure out what potential tags there are, and then generate the usual `dist plan` output for each one.

Possibly something closer to this:

```
  × There are too many unrelated apps in your workspace to coherently Announce!
  help: Please either specify --tag, or give them all the same version
        
        Here are some options:
        
        --tag=v1.0.0 will Announce: foo, bar
        --tag=v2.0.0 will Announce: baz
        
        you can also request any single package with --tag=foo-v1.0.0

--tag=v1.0.0 would announce
  foo 1.0.0
    [...]
  bar 1.0.0
    [...]

--tag=v2.0.0 would announce
  baz 2.0.0
    [...]
```

**Comment by duckinator** [comment]
I'm not entirely sure that we want to keep the exact "There are too many ..."/"help: ..." text we have now, but I'm also not sure what to replace it with. I think the information it's conveying is useful, but may be a bit confusing when combined with information on what `--tag=...` options can be used.

There is also the open question of how to handle the JSON output.

**Comment by kekonn** [comment]
I am running into a similar issue, where dist plan is being run as a PR check (I believe this is how cargo dist sets itself up?) and because a PR does not have a specific tag, I couldn't even apply one if I wanted to.

So in short. Doing this check when running as part of a PR check, seems like a bug to me.

**Comment by duckinator** [comment]
I just went through all the notes I have and what @alilleybrinker has mentioned, and I boiled it all down to 3 chunks:

1. [ ] if `pr-run-mode = "plan"`: Set `needs_coherence = false` in `cmd_plan()`. (See [@alilleybrinker's work](https://github.com/alilleybrinker/cargo-dist/pull/1).)
2. [ ] if `pr-run-mode = "upload"`: Have `dist build` run for every discovered tag.
3. [ ] if there's more than one tag, have `dist plan` show more information, e.g. #1560.

The first one being the most important, since it is actively causing problems for multiple users.

With just the first one resolved, mismatched tags/versions + the default `pr-run-mode = "plan"` will work, but mismatched tags/versions + `pr-run-mode = "upload"` will NOT work.
